### PR TITLE
Feature/17317 accessibility issues on show more pattern for provider selection

### PR DESCRIPTION
**Pull request by jenniemc**
## Description
When clicking on the "more provider" link it currently does not focus first next set of providers. The same is also missing from the "more locations" link in the facility selection page

## Testing done
local and unit test

## Screenshots
image

## Acceptance criteria
- [ ] have focus on the first next set of providers
- [ ] add aria-label as needed
- [ ] add .sr-only to visually hide text from screen but read out loud by screen reader

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 1]
```diff
@@ -300,15 +326,20 @@ function ProviderSelectionField({
             {providersListLength < communityCareProviderList.length && (
               <>
                 <button
+                  aria-atomic="true"
+                  aria-live="assertive"
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:330]: Seems like this would need to have `aria-live="polite" role="status" aria-atomic="true"` instead of these two attributes.

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 2]
```diff
@@ -127,7 +143,11 @@ function ProviderSelectionField({
               )}{' '}
               miles
             </span>
-            <div className="vads-u-display--flex">
+            <div
+              className="vads-u-display--flex"
+              aria-atomic="true"
+              aria-live="assertive"
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:149]: Why was this added?

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx [hunk 3]
```diff
@@ -24,20 +25,29 @@ export default function FacilitiesRadioWidget({
...
+    },
+    [displayedOptions.length, displayAll],
+  );
 
   return (
     <div>
-      {displayedOptions.map((option, i) => {
-        const { id, name, address, legacyVAR } = option?.label;
-        const checked = option.value === value;
+      {displayedOptions.map((facility, facilityIndex) => {
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:48]: I would probably leave the first one named `option`. It has some facility data in it, but the structure is more of an option object, which in the context of our forms means an object with a `value` and `label` property.

The second one you could leave as `i` or change to `optionIndex`.

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx [hunk 4]
```diff
@@ -47,19 +57,25 @@ export default function FacilitiesRadioWidget({
...
         }
+        const facilityPosition = facilityIndex + 1;
 
         return (
-          <div className="form-radio-buttons" key={option.value}>
+          <div
+            className="form-radio-buttons"
+            key={facility.value}
+            aria-atomic="true"
+            aria-live="assertive"
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:67]: Same question as before, not sure why we're adding these aria-live attributes

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx [hunk 5]
```diff
@@ -47,19 +57,25 @@ export default function FacilitiesRadioWidget({
...
+          >
             <input
               type="radio"
               checked={checked}
-              id={`${id}_${i}`}
+              id={`facility_${facilityPosition}`}
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:72]: Same question as the previous page here, too, I'd like to stick with the `id` prop instead of facility in the first part of this string, if possible.

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx [hunk 6]
```diff
@@ -81,8 +97,13 @@ export default function FacilitiesRadioWidget({
           <button
             type="button"
             className="additional-info-button va-button-link vads-u-display--block"
-            onClick={() => setDisplayAll(!displayAll)}
+            aria-atomic="true"
+            aria-live="assertive"
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:101]: I would remove these two aria attributes entirely here, since this button text will never change. It'll be removed once it is clicked.

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 7]
```diff
@@ -241,18 +261,24 @@ function ProviderSelectionField({
...
                   const { name } = provider;
                   const checked = provider.id === checkedProvider;
+                  const providerPosition = providerIndex + 1;
                   return (
-                    <div className="form-radio-buttons" key={provider.id}>
+                    <div
+                      className="form-radio-buttons"
+                      key={provider.id}
+                      aria-atomic="true"
+                      aria-live="assertive"
```
[the remark this answers is not in the document]
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:270]: Deleted

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 8]
```diff
@@ -241,18 +261,24 @@ function ProviderSelectionField({
...
+                    >
                       <input
                         type="radio"
                         checked={checked}
-                        id={`${idSchema.$id}_${provider.id}`}
+                        id={`provider_${providerPosition}`}
```
[the remark this answers is not in the document]
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:275]: replaced  id={`provider_${providerPosition}`} with id={`${idSchema.$id}_${providerPosition}`}

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 9]
```diff
@@ -276,7 +302,7 @@ function ProviderSelectionField({
                             setShowProvidersList(false);
                             recordEvent({
                               event: `${GA_PREFIX}-order-position-provider-selection`,
-                              providerPosition: providerIndex + 1,
+                              providerPosition: { providerPosition },
```
[the remark this answers is not in the document]
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:305]: Done

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx (hunk 1 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:330]: switched `aria-live="polite"` but cannot add `role="status"` because eslint shows "Interactive elements should not be assigned non-interactive roles."

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx (hunk 2 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:149]: Deleted

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx (hunk 4 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:67]: Deleted

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx (hunk 6 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:101]: Deleted

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx (hunk 3 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:48]: reverted back to `option` and `i`

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx (hunk 5 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:72]: replaced id={`facility_${facilityPosition}`} with id={`${id}_${facilityPosition}`}

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 10]
```diff
@@ -234,25 +250,26 @@ function ProviderSelectionField({
                     </button>
                   </p>
                 )}
-                <p>
+                <p role="status">
```
[the remark this answers is not in the document]
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx]: Added aria attributes

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 11]
```diff
@@ -300,15 +317,20 @@ function ProviderSelectionField({
             {providersListLength < communityCareProviderList.length && (
               <>
                 <button
+                  aria-live="polite"
+                  aria-atomic="true"
```
[the remark this answers is not in the document]
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx]: Removed aria attributes from button tag

**Review by jbalboni — APPROVED** [review]
